main/src/ca/uwaterloo/flix/language/phase/typer/ConstraintGen.scala

```diff
@@ -743,8 +743,14 @@ object ConstraintGen {
 
         (resTpe, resEff)
 
-      case Expr.InvokeMethod2(obj, name, exps, tvar, evar, loc) =>
-        ??? // TODO
+      case Expr.InvokeMethod2(exp, name, exps, tvar, evar, loc) =>
+        val (tpe, eff) = visitExp(exp)
+        val (tpes, effs) = exps.map(visitExp).unzip
+        c.unifyType(tvar, Type.mkApply(Type.Cst(TypeConstructor.MethodReturnType(name.name, exps.length), loc), tpe :: tpes, loc), loc)
```

Comment:
Lets introduce local (Scala) variables for Type.mkApply(...) and Type.mkUnion(...)

We can call them t and r.